dashboard/app/models/script.rb

```diff
@@ -1021,7 +1021,7 @@ def self.base_name(name)
   def copy_and_write_i18n(new_name)
     scripts_yml = File.expand_path('config/locales/scripts.en.yml')
     i18n = File.exist?(scripts_yml) ? YAML.load_file(scripts_yml) : {}
-    i18n.deep_merge!(summarize_i18n_for_copy(new_name)) {|_, old, _| old}
+    i18n.deep_merge!(summarize_i18n_for_copy(new_name))
```

Comment:
why are we removing the block passed to `deep_merge!` here? I'm not saying it's wrong, but it looks a bit tricky and related to a thing I need to do to the summarize methods next so hoping to learn something.

Reply:
So removing the {|_, old, _| old} will make it so if there is the same key in both i18n and summarize_i18n_for_copy it will default to the one in summarize_i18n_for_copy. That seemed to me like what we would want but you know this area of the code better than I do so maybe thats incorrect?

Reply:
My concern was that this change might expose some bug like the one we saw where the i18n system did not immediately reflect a new value after one was written, i.e. the scripts.en.yml file would be guaranteed to be fresh but whatever you got from summarize_i18n_for_copy might not be because it calls `I18n.t`. 

I dug in github commit history and ended up with this gem: https://github.com/code-dot-org/code-dot-org/commit/4cb8b739fef022d4f605a5fbe87586cf935e938c the commit description points to 2 other commits which flip flop between using or not using this block, both of which were merged to staging back in 2014 without a PR and without comment! It really was the wild west back then :-)

So based on all of that I don't want to give any special credit to the way the code was, although it still needs a small amount of credit since we know that it's been working for about 4 years.

I would say, go ahead and make this change. If i am reading correctly, this code line won't have any effect most of the time, and I think your change may actually fix a problem by updating scripts.en.yml to actually properly copy of all the i18n when we copy a script. we can verify that when we focus on script copy in December or January.